src/wms/src/main/java/org/geoserver/wms/decoration/LegendDecoration.java

```diff
@@ -293,4 +313,55 @@ private String findTitle(Layer layer, Catalog catalog) {
         public BufferedImage legend;
         public GetLegendGraphicRequest request;
     }
+
+    private LegendInfo resolveLegendInfo(
+            LegendInfo legendInfo, GetLegendGraphicRequest request, StyleInfo context) {
+        if (legendInfo == null) {
+            return null; // not available
+        }
+        String onlineResource = legendInfo.getOnlineResource();
+        String baseUrl = request.getBaseUrl();
+        if (onlineResource == null) {
+            return null;
+        }
+        URL url = null;
+        try {
+            URI uri = new URI(onlineResource);
+            GeoServerResourceLoader resources = wms.getCatalog().getResourceLoader();
+            if (uri.isAbsolute()) {
```

Comment:
This resolution dance, is this the only place where it needs to happen? No usage in UI for example?

Reply:
No indeed it happens in the GetLegendGraphicKvpReader class. I change the code to use that method instead.